### PR TITLE
[core] Don't update destination in blob extraction if blob is null

### LINKED ISSUE #5342: 🐛 Server Crash upon New Char Create
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my issue will be ignored.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have searched existing [issues](https://github.com/LandSandBoat/server/issues) to see if the issue has already been opened, and I have checked the commit log to see if the issue has been resolved since my server was last updated.

## OS / platform the server is running (if known)

Windows 11

## Branch affected by issue 

`base`

## Steps to reproduce

With the newest update from the game. When creating a new char and attempt log in after picking starting area, server will crash. Char does save in the Database, When attempt log in again after selecting that char server will crash again. Can log into the Char on Older update just fine and no issues. I have tested this on Two different Window PC's with both updated on latest. I can log in on my main Char on the newest update fine without a server crash, it's just for anyone that is new to the server.

## Expected behavior

Not have the server crash upon logging in a new created char.

### PR DESCRIPTION
**Pull request by WinterSolstice8**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

Fixes #5342

## Steps to test these changes

Create a new character, login without crash

### CODE REVIEW

**Review by zach2good — APPROVED** [review]
:thinking: huh, but I literally tested new character creation and log in / log out / log in at least 4 times when I was testing alliance and cross process stuff

Even if we're not copying the contents of the null string (which I assume is the issue?), should we be memsetting the target struct, because we've asked to load the contents of the db into that struct anyway? To me this is never additive, it should always be destructive/replacement

**Comment by zach2good** [comment]
To the effect of:

```cpp
        std::memset(&destination, 0x00, sizeof(T));
        if (!rset->isNull(blobKey.c_str())) // This can do very bad things if the blob is NULL.
        {
            auto blobStr = rset->getString(blobKey.c_str());
            std::memcpy(&destination, blobStr.c_str(), sizeof(T));
        }
```

I should imagine

**Comment by zach2good** [comment]
And maybe reword the comment to be clearer about what will happen: something something reading in uninitialised memory which will populate `destination` with garbage, leading to difficult to diagnose crashes etc.

I like using the terminology "difficult to diagnose crashes" because it puts the fear of god into anyone who touches compilers.

**Comment by zach2good** [comment]
Assuming it was before/after, and that the eminence blob was zero'd out before use and didn't start as uninit'd garbage

**Comment by zach2good** [comment]
I think cpp changes from web mess up CI, can you fix this for me when you get the chance 🙏

**Comment by WinterSolstice8** [comment]
> I think cpp changes from web mess up CI, can you fix this for me when you get the chance 🙏

Seems to be an extra newline, I'm out for a bit but will do

**Review by Xaver-DaRed — APPROVED** [review]

**Review by zach2good — APPROVED** [review]
